src/sobol_sensitivity.jl

```diff
@@ -346,6 +346,7 @@ function gsa(f, method::Sobol, p_range::AbstractVector; samples, kwargs...)
     AB = QuasiMonteCarlo.generate_design_matrices(samples, [i[1] for i in p_range],
         [i[2] for i in p_range],
         QuasiMonteCarlo.SobolSample(),
+        QuasiMonteCarlo.OwenScramble(),
```

Comment:
My understanding based on https://docs.sciml.ai/QuasiMonteCarlo/stable/randomization/#Scrambling-methods is that you have to specify
```suggestion
        QuasiMonteCarlo.OwenScramble(; base=2, pad=ceil(Int, log2(samples))),
```
(or some larger `pad`).
Additionally, I think we would want to forward an `rng` here if possible:
```suggestion
        QuasiMonteCarlo.OwenScramble(; base=2, pad=ceil(Int, log2(samples)), rng),
```

Given the comments in QuasiMonteCarlo, I also wonder if one should use `MatousekScramble` instead. But since generation of the design matrix shouldn't be the bottleneck here maybe it's good to stick to the well studied `OwenScramble` and the performance gains from `MatousekScramble` are not worth it.